### PR TITLE
Chapter/chapter1 control structures and function

### PR DESCRIPTION
**Pull request by hothanhtuan1209**
I create this pull request to merge code in chapter 1 to branch develop

### CODE REVIEW

File: Control_Structures_and_Function/Exceptions/useful_exceptions.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+"""
+- Avoid “swallowing” useful exceptions with bare except clauses.
+- Programmers often make the mistake of overly relying on generic exception handling, especially when using third-party packages.
+- This leads to a loss of crucial debugging information.
+"""
+
+# You shouldn't do like this
+import requests
```
Comment (cuongdoan-agilityio) [inline Control_Structures_and_Function/Exceptions/useful_exceptions.py]: The import statement should be at the top of the file and after the docstring.

File: Control_Structures_and_Function/Exceptions/use_exceptioons_EAFP.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+"""
+- Use Exceptions to write code in an “EAFP” style
+- Two programming styles: "Look Before You Leap (LBYL)," which involves extensive pre-condition checks. 
+- And "Easier to Ask for Forgiveness than Permission (EAFP)," which focuses on catching exceptions when things go wrong, emphasizing code clarity and efficiency.
+"""
+
+# You shouldn't do like this
```
Comment (cuongdoan-agilityio) [inline Control_Structures_and_Function/Exceptions/use_exceptioons_EAFP.py]: `Harmful solution`

File: Control_Structures_and_Function/Exceptions/use_exceptioons_EAFP.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        elif not "DEFAULT_LOG_LEVEL" in config_dict:
+            return None
+        else:
+            return config_dict["DEFAULT_LOG_LEVEL"]
+
+    else:
+        return None
+
+
+# You should do like this
```
Comment (cuongdoan-agilityio) [inline Control_Structures_and_Function/Exceptions/use_exceptioons_EAFP.py]: `Idiomatic solution`

File: Control_Structures_and_Function/Exceptions/use_exceptioons_EAFP.py (hunk 2 above)
Comment (hothanhtuan1209) [reply Control_Structures_and_Function/Exceptions/use_exceptioons_EAFP.py]: Done

File: Control_Structures_and_Function/Exceptions/use_exceptioons_EAFP.py (hunk 3 above)
Comment (hothanhtuan1209) [reply Control_Structures_and_Function/Exceptions/use_exceptioons_EAFP.py]: Done

File: Control_Structures_and_Function/Exceptions/useful_exceptions.py (hunk 1 above)
Comment (hothanhtuan1209) [reply Control_Structures_and_Function/Exceptions/useful_exceptions.py]: Done

File: Control_Structures_and_Function/Functions/use_return_to_evaluate.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+# Harmful solution
+# def all_equal(a, b, c):
+#     result = False
+#     if a == b == c:
+#         result = True
+#     return result
+
+
+# Idiomatic solution
+def all_equal(a, b, c):
```
Comment (cuongdoan-agilityio) [inline Control_Structures_and_Function/Functions/use_return_to_evaluate.py]: Should update the parameters name.

**Comment by cuongdoan-agilityio** [comment]
Control_Structures_and_Function/Exceptions/use_exceptioons_EAFP.py
The file name does not contain capital letters.

File: Control_Structures_and_Function/Functions/avoid_default_value.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+#     return L
+
+# print(f(1))
+# print(f(2))
+# print(f(3))
+
+
+# Idiomatic solution
+def f(a, L=None):
```
Comment (cuongdoan-agilityio) [inline Control_Structures_and_Function/Functions/avoid_default_value.py]: Update function and parameters name.

File: Control_Structures_and_Function/Functions/use_args_kwargs.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    
+    def wrapper(*args, **kwargs):
+        print("--------------------------------")
+        print(str(func(*args, **kwargs)))
+        print("--------------------------------")
+
+    return wrapper
+
+@for_console_output
+def add(x, y):
```
Comment (cuongdoan-agilityio) [inline Control_Structures_and_Function/Functions/use_args_kwargs.py]: Update parameters name.

File: README.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+## Required
+- To run code, you need to install python version 3.11 or higher and install the environments to use python.
+
+## How to use
+1. First, clone this repository on your computer
+> git clone https://github.com/hothanhtuan1209/Write-Idiomatics-Python.git
+
+2. Checkout to branch develop
+> git checkout develop
+
```
Comment (cuongdoan-agilityio) [inline README.md:18]: How to run your codes?

File: Control_Structures_and_Function/Functions/use_return_to_evaluate.py (hunk 4 above)
Comment (hothanhtuan1209) [reply Control_Structures_and_Function/Functions/use_return_to_evaluate.py]: Done

File: Control_Structures_and_Function/Functions/avoid_default_value.py (hunk 5 above)
Comment (hothanhtuan1209) [reply Control_Structures_and_Function/Functions/avoid_default_value.py]: Done

File: Control_Structures_and_Function/Functions/use_args_kwargs.py (hunk 6 above)
Comment (hothanhtuan1209) [reply Control_Structures_and_Function/Functions/use_args_kwargs.py]: Done

**Pull request merged**